python/pyarrow/table.pxi

```diff
@@ -4758,11 +4758,11 @@ cdef class Table(_PandasConvertible):
         if isinstance(sorting, str):
             sorting = [(sorting, "ascending")]
 
-        indices = _pc().sort_indices(
-            self,
-            options=_pc().SortOptions(sort_keys=sorting, **kwargs)
-        )
-        return self.take(indices)
+        res = _pc()._exec_plan._sort_source(self, output_type=Table,
+                                            sort_options=_pc().SortOptions(
+                                                sort_keys=sorting, **kwargs
+                                            ))
```

Comment:
Is there any advantage is using the exec plan version here, instead of the current version? (but I suppose also no disadvantage)

Reply:
I didn't benchmark it, I can if you want to have numbers, but I guessed that it's faster to do the whole operation in C++ once than having to sort indices and create a temporary arrays of indices just to then take the actual values from the table.

Reply:
For future reference, at the time I replaced this with a manual sort_indices call in a follow-up PR (https://github.com/apache/arrow/pull/15268, for enabling this in a minimal build). But so it also appears that the Acero-based implementation is currently actually slower, see https://github.com/apache/arrow/pull/34249#issuecomment-1468028906 for some analysis (so as long as that is not solved, eg by allowing configuring the batch size, we should keep this manual implementation for Table/RecordBatch.sort_by).